src/azul/transformer.py

```diff
@@ -300,6 +300,43 @@ def get(self):
         return thaw(super().get())
 
 
+class FrequencySetAccumulator(Accumulator):
+    """
+    An accumulator that accepts any number of values and returns a list that has
+    length max_size or smaller and is ordered by most frequent value to less frequent.
+
+    >>> a = FrequencySetAccumulator(2)
+    >>> a.accumulate('x')
+    True
+    >>> a.accumulate(['x','y'])
+    True
+    >>> a.accumulate({'x','y','z'})
+    True
+    >>> a.get()
+    ['x', 'y']
+    >>> a = FrequencySetAccumulator(0)
+    >>> a.accumulate('x')
+    True
+    >>> a.get()
+    []
+    """
+    def __init__(self, max_size) -> None:
+        super().__init__()
+        self.value = Counter()
+        self.max_size = max_size
+
+    def accumulate(self, value) -> bool:
+        original_count = sum(self.value.values())
+        if isinstance(value, (list, set)):
+            self.value.update(value)
+        else:
+            self.value[value] += 1
+        return original_count < sum(self.value.values())
+
+    def get(self) -> List[Any]:
```

Comment:
```suggestion
    def get(self) -> Set[Any]:
```

Reply:
Returning a set is problematic later on in the code
```
Error
Traceback (most recent call last):
  File "/Users/daniel/azul/.venv/lib/python3.6/site-packages/elasticsearch/serializer.py", line 48, in dumps
    return json.dumps(data, default=self.default, ensure_ascii=False)
  File "/usr/local/Cellar/python/3.6.5_1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/json/__init__.py", line 238, in dumps
    **kw).encode(obj)
  File "/usr/local/Cellar/python/3.6.5_1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/local/Cellar/python/3.6.5_1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/Users/daniel/azul/.venv/lib/python3.6/site-packages/elasticsearch/serializer.py", line 34, in default
    raise TypeError("Unable to serialize %r (type: %s)" % (data, type(data)))
TypeError: Unable to serialize {'in situ sequencing'} (type: <class 'set'>)
```

Reply:
Got it.